docs/runtimes/node-affinity.md

```diff
@@ -1,96 +1,347 @@
+  of the function, and select **Edit** and scroll to the **Resources** section.
+-->
 
 **In this section**
-- [On demand vs spot](#On-demand-vs-spot)
-- [Stateless and stateful applications](#Stateless-and-stateful-applications)
-- [Node selector](#node-selector)
+- [Replicas](#replicas)
+- [CPU, GPU, and memory limits for user jobs](#cpu-gpu-and-memory-limits-for-user-jobs)
+- [Volumes](#volumes)
+- [Preemption mode: Spot vs. On-demand nodes](#preemption-mode-spot-vs-on-demand-nodes)
+- [Pod priority](#pod-priority-for-user-jobs)
+- [Node affinity (node selectors)](#node-affinity-node-selectors)
+
+## Replicas
+
+Some runtimes can scale horizontally, configured either as a number of replicas:</br>
+`spec.replicas` </br>
+or a range (for auto scaling in Dask or Nuclio:</br>
+```
+spec.min_replicas = 1
+spec.max_replicas = 4
+```
+See more details in [Dask](../runtimes/dask-overview), [MPIJob and Horovod](../runtimes/horovod), [Spark](../runtimes/spark-operator), [Nuclio](../concepts/nuclio-real-time-functions).
 
-## On demand vs spot 
+## CPU, GPU, and memory limits for user jobs  
 
-Amazon Elastic Compute Cloud (Amazon EC2) provides scalable computing capacity in the Amazon Web Services (AWS) Cloud. 
-Using Amazon EC2 eliminates your need to invest in hardware up front, so you can develop and deploy applications faster. 
+When you create a pod in an MLRun job or Nuclio function, the pod has default CPU and memory limits. When the job runs, it can consume 
+resources up to the limits defined. The default limits are set at the service level. You can change the default limit for the service, and 
+also overwrite the default when creating a job, or a function. 
+<!-- The default values are: -->
 
-Using the Iguazio platform you can deploy two different kinds of EC2 instances, on-demand and spot. 
-On-Demand Instances provide full control over the EC2 instance lifecycle. You decide when to launch, stop, hibernate, start, 
-reboot, or terminate it. With spot instances you request EC2 capacity from specific availability zones and is 
-susceptible to spot capacity availability. This is a good choice if you can be flexible about when your applications run 
-and if your applications can be interrupted.
+See more about [Kubernetes Resource Management for Pods and Containers](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/).
+
+### UI configuration
+When creating a service, set the **Memory** and **CPU** in the **Common Parameters** tab, under **User jobs defaults**.
+When creating a job or a function, overwrite the default **Memory**, **CPU**, or **GPU** in the **Configuration** tab, under **Resources**.
+
+### SDK configuration
+
+Configure the limits assigned to a function by using `with_limits`. For example:
+
+```
+import mlrun
+import os
+fn = mlrun.import_function("hub://xgb_trainer")
```

Comment:
not a good example, this market func will soon be deprecated and too complicated, see cells 8 & 9 in: https://github.com/mlrun/demos/blob/1.0.x/mask-detection/1-training-and-evaluation.ipynb

Reply:
This section should show example of GPU, limits, and memory. The one you pointed to is only GPUs. How to add memory and limits to this example?
# If you wantto train on gpu, set this variable to 'True', otherwise 'False':
use_gpu = False

# Create the MLRun Function:
mpi_training_and_evaluation_function = mlrun.code_to_function(
    filename=os.path.join(framework, "training-and-evaluation.py"),
    name="mpi-training-and-evaluation",
    handler="train",
    kind="mpijob",
    image="mlrun/ml-models-gpu" if use_gpu else "mlrun/ml-models",
    with_doc=False
)

# Setup the desired configurations:
mpi_training_and_evaluation_function.spec.replicas = 2
if use_gpu:
    # Select the number of GPUs per replica:
    mpi_training_and_evaluation_function.gpus(1)
else:
    mpi_training_and_evaluation_function.with_requests(cpu=2)

Reply:
@jillnogold keep it simple:

```python
training_function = mlrun.code_to_function("training.py", name="training", handler="train", 
                                                                       kind="mpijob", image="mlrun/ml-models-gpu")
training_function.spec.replicas = 2
training_function.with_requests(cpu=2)
training_function.gpus(1)
```